config/source/file.go

```diff
@@ -0,0 +1,134 @@
+package source
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-spatial/tegola/internal/env"
+	"github.com/go-spatial/tegola/internal/log"
+)
+
+// FileConfigSource is a config source for loading and watching files in a local directory.
+type FileConfigSource struct {
+	dir string
+}
+
+func (s *FileConfigSource) Type() string {
+	return "file"
+}
+
+func (s *FileConfigSource) Init(options env.Dict, baseDir string) error {
+	var err error
+	dir, err := options.String("dir", nil)
+	if err != nil {
+		return err
+	}
+
+	// If dir is relative, make it relative to baseDir.
+	if !filepath.IsAbs(dir) {
+		dir = filepath.Join(baseDir, dir)
```

Comment:
We probably should do a filepath.Clean here as well?

Reply:
`filepath.Join` calls `Clean` internally. https://pkg.go.dev/path/filepath#Join